### DISCUSSION TITLE
Circle CI checks are stuck in pending with Merge Queue and are not ran

### DISCUSSION THREAD

**Discussion by john-kajabi** [community/community#38199] [Pull Requests]
Hi all,

Currently we are beta testing the Merge Queue feature with CircleCI. We are noticing that our PR's are stuck in the queue because our CircleCI checks are not ran. Here is an example of where our PRs are in the queue for 19 hours with no status checks failing, just stuck in pending.

<img width="1265" alt="image" src="https://user-images.githubusercontent.com/89809992/199780981-589889c2-18ef-4f05-9f35-f1a95daf7368.png">

<img width="639" alt="image" src="https://user-images.githubusercontent.com/89809992/199781083-82600361-cdac-4a02-8f4c-0101449ea515.png">

I'm not sure why these checks wouldn't be ran. They run when the PR is initially created before we add it to the merge queue and must pass, otherwise it would fail branch protection rules and we wouldn't be able to it to the merge queue at all. Thanks in advance for the support!

**Comment by antonblr** [comment]
Hi @john-kajabi - I'd refer to the CircleCI documentation to debug that. 

1) First ensure that everything is setup as per https://circleci.com/docs/github-integration/#overview. Specifically verify these two items are set in the GitHub repo settings:
   * A deploy key that is used to check out your project from GitHub.
   * A service hook (or "push hook") that is used to notify CircleCI when you push to GitHub.
  2) Another possibility is described [Troubleshooting – GitHub Checks waiting for status in GitHub](https://circleci.com/docs/enable-checks/#troubleshooting-github-checks-waiting-for-status-in-github) article (that if you have GitHub Checks enabled. Not to confuse with GitHub status updates)
3) And finally, you might want to re-do the project setup in CCI:
   * Remove all CircleCI webhooks and services from your GitHub repo settings.
   * Unfollow the project on CircleCI through “Project settings”.
   * Setup / Re-follow the project on CircleCI.

**Reply by Rochebeard** [reply]
Rocje
- [ ]

**Comment by medghaim** [comment]
For people looking for a solution to this, there's a good chance that both your GitHub & CircleCI is misconfigured.

**Github Webhook Configuration**
When configuring your Github -> CircleCI webhook (in Github webhook settings), if you have "Let me select individual events" selected:
![image](https://user-images.githubusercontent.com/7801031/211115130-cfcfdfc4-63d4-4aa5-8541-647cd293b924.png)

Then make sure you have the following setting selected from the list of events:
![image](https://user-images.githubusercontent.com/7801031/211115366-59b45344-9ab0-49f4-9820-21a11a58f7d6.png)

**CircleCI**
If your project has "Only build pull requests", then your pipeline will only get triggered when Github sends a `pull_request` event to the CircleCI webhook. The problem with this is that when adding to a Github merge queue, it sends a `merge_group` event, which CircleCI ignores.

From [CircleCI documentation](https://circleci.com/docs/github-integration/#overview):
> Currently, the only way to override the Only build pull requests setting is to open a support request at https://support.circleci.com/. In the request, please specify the regular expression(s) that you would like to add to the allow-list of branches for which CircleCI will validate every commit.

Thankfully, the temporary branch that github creates for your merge queue is prepended with `gh-readonly-queue/`. So, we need to reach out to CircleCI support and ask them to add this regular expression to your projects `allow-list`:
```
gh\-readonly\-queue.\*
```

**Comment by e-grapp** [comment]
Howdy folks - checking in from CircleCI support here!

We have recently enabled users to be able to apply these overrides at the project level for any projects leveraging the merge queue and encountering this behavior.

#### Announcement: 

> https://discuss.circleci.com/t/product-launch-override-only-build-prs-setting/47807/6 

#### Article on How to apply these overrides:

> https://support.circleci.com/hc/en-us/articles/15222074173723-How-to-add-additional-Default-Branches

Hope this helps unblock any folks who experience this behavior!
